File: 卡码/代码随想录/Python/哈希/04-两数之和/LeetCode_0001.py
#1. 两数之和
from typing import *

class Solution:
	def twoSum(self, nums: List[int], target: int) -> List[int]:
		map=dict()
		for index,value in enumerate(nums):
			if target-value in map:
				return [index,map[target-value]]
			else:
				map[value]=index
		return []

File: 卡码/代码随想录/C++/哈希/04-两数之和/LeetCode_0001.cpp
//1. 两数之和
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
  	unordered_map<int,int> m;
  	for(int i=0;i<nums.size();i++){
			if(m.find(target-nums[i])!=m.end()) return vector<int>{m[target-nums[i]],i};
			else m.insert(make_pair(nums[i],i));
		}
		return vector<int>{};
  }
};
